p2p/protocol/identify/id.go

```diff
@@ -238,11 +238,8 @@ func (ids *IDService) consumeMessage(mes *pb.Identify, c inet.Conn) {
 		lmaddrs = append(lmaddrs, maddr)
 	}
 
-	// if the address reported by the connection roughly matches their annoucned
-	// listener addresses, its likely to be an external NAT address
-	if HasConsistentTransport(c.RemoteMultiaddr(), lmaddrs) {
-		lmaddrs = append(lmaddrs, c.RemoteMultiaddr())
-	}
+	// NOTE: Do not add `c.RemoteMultiaddr()` to the peerstore if the remote
+	// peer doesn't tell us to do so. Otherwise, we'll advertise it.
```

Comment:
@whyrusleeping 

Proposal: Don't even _record_ the remote peer's address in the peerstore if they aren't advertising it.
Rational: If we record it, we'll share it. I'm worried this is causing us to gossip bad ephemeral addresses around the network. I'm still seeing some addr-splosion peers.

Objections?

Reply:
SGTM.

Reply:
This seems like a good idea, though I hope we're not forgetting the reason we added these in the first place.

I can't think of any good reason to be doing this right now, but let's make sure that the rationale for this decision is well documented (including what happens when we advertise these addresses without the authors consent)

Reply:
I've extended the comment with more rational so we don't run into this situation again.